common/config_parser/config_parser.cpp

```diff
@@ -58,20 +57,19 @@ ExpectedBool MenderConfigFromFile::ValidateArtifactKeyCondition() const {
 }
 
 ExpectedBool MenderConfigFromFile::ValidateServerConfig() const {
-	auto conf = *this;
-	if (conf.server_url.size() != 0 && conf.servers.size() != 0) {
+	if (server_url.size() != 0 && servers.size() != 0) {
 		auto err = MakeError(
 			ConfigParserErrorCode::ParseError,
 			"Both 'Servers' AND 'ServerURL given in the configuration. Please set only one of these fields");
-		return ExpectedBool(err);
+		return err;
 	}
 
-	if (conf.servers.size() == 0) {
-		if (conf.server_url.size() == 0) {
+	if (servers.size() == 0) {
+		if (server_url.size() == 0) {
 			// TODO - Log warning
 		}
 	}
-	return ExpectedBool(true);
+	return true;
```

Comment:
We should replace the return type with `error::Error` actually, and then return `error::NoError`. Returning `ExpectedBool` is a leftover from before we adopted that method.

But this requires changes in the caller as well, and is unrelated to this PR, so I think this can be a followup.

In fact, we should do that pretty much everywhere where we use `ExpectedBool` except if there is a place where we actually use the boolean value (but I don't think there is).

Reply:
Agree, probably 'ExpectedBool Json::GetBool() const " is the only place we should use ExpectedBool